### ISSUE TITLE
demo_faster_rcnn.py shape incompatible expected(5,0) vs saved(21, 2048)

### ISSUE THREAD

**Issue by LoryMANG** [dmlc/gluon-cv#1318]
When i run the demo_faster_rcnn.py to test my own trained model. It occurs an Error: failed loading parameter fasterrcnn0_dense0_weight from saved params:shape incompatible expected(5,0) vs saved(21, 2048)
Is there any way to solve this?

**Comment by LoryMANG** [comment]
`import time

from gluoncv.data.transforms import presets

from gluoncv import model_zoo, data, utils
#from matplotlib import pyplot as plt
import mxnet as mx
import cv2
import argparse
import os

def parse_args():
    parser = argparse.ArgumentParser(description='Train faster rcnn networks with random input shape.')
    parser.add_argument('--network', type=str, default='faster_rcnn_resnet50_v1b_voc',
                        help="Base network name which serves as feature extraction base.")
    parser.add_argument('--short', type=int, default=416,
                        help='Input data shape for evaluation, use 320, 416, 512, 608, '                  
                        'larger size for dense object and big size input')
    parser.add_argument('--threshold', type=float, default=0.4,
                        help='confidence threshold for object detection')

    parser.add_argument('--gpu', action='store_false',
                        help='use gpu or cpu.')

    args = parser.parse_args()
    return args


if __name__ == '__main__':
    args = parse_args()
    if args.gpu:
        ctx = mx.gpu()
    else:
        ctx = mx.cpu()

    net = model_zoo.get_model(args.network, pretrained=False)

    # classes = ['shoe']
    # for param in net.collect_params().values():
    #     if param._data is not None:
    #         continue
    #     param.initialize()
    # net.reset_class(classes)
    # net.collect_params().reset_ctx(ctx)

    if args.network == 'faster_rcnn_resnet50_v1b_voc':
        net.load_parameters('./model_fasterrcnn/faster_rcnn_resnet50_v1b_voc_best.params',ctx=ctx)
        print('use faster_rcnn_resnet50_v1b_voc to extract feature')
    else:
        net.load_parameters('./model_resnet101/faster_rcnn_resnet101_v1d_voc_best.params',ctx=ctx)
        print('use faster_rcnn_resnet101_v1d to extract feature')

    picpath = "test_pics"
    time_start = 0
    for i in os.listdir(picpath):
        frame = os.path.join(picpath + '/' + i)
        savepath = "result_pics/" + i.split('.')[0] + '_result.jpg'      # xxx_result.jpg

    # frame = '/home/Safety-Helmet-Wearing-Dataset-master/image/20200403131100_2764.jpg'
        parttime = time.time()
        # x, orig_img = data.transforms.presets.rcnn.load_test(frame, short=args.short)
        x, orig_img = presets.rcnn.load_test(frame, short=args.short)
        x = x.as_in_context(ctx)
        box_ids, scores, bboxes = net(x)
        ax = utils.viz.cv_plot_bbox(orig_img, bboxes[0], scores[0], box_ids[0], class_names=net.classes,thresh=args.threshold)
        time_start += time.time()-parttime
        # cv2.imshow('image', orig_img[...,::-1])
        # cv2.waitKey(0)
        cv2.imwrite(savepath, orig_img[...,::-1])
        # cv2.destroyAllWindows()
    # totaltime = time.time()-time_start
    print("time is :", time_start)`

**Comment by LoryMANG** [comment]
`import time

from gluoncv.data.transforms import presets

from gluoncv import model_zoo, data, utils
import mxnet as mx
import cv2
import argparse
import os

def parse_args():
    parser = argparse.ArgumentParser(description='Train faster rcnn networks with random input shape.')
    parser.add_argument('--network', type=str, default='faster_rcnn_resnet50_v1b_voc',
                        help="Base network name which serves as feature extraction base.")
    parser.add_argument('--short', type=int, default=416,
                        help='Input data shape for evaluation, use 320, 416, 512, 608, '                  
                        'larger size for dense object and big size input')
    parser.add_argument('--threshold', type=float, default=0.4,
                        help='confidence threshold for object detection')

    parser.add_argument('--gpu', action='store_false',
                        help='use gpu or cpu.')

    args = parser.parse_args()
    return args


if __name__ == '__main__':
    args = parse_args()
    if args.gpu:
        ctx = mx.gpu()
    else:
        ctx = mx.cpu()

    net = model_zoo.get_model(args.network, pretrained=False)

    # classes = ['shoe']
    # for param in net.collect_params().values():
    #     if param._data is not None:
    #         continue
    #     param.initialize()
    # net.reset_class(classes)
    # net.collect_params().reset_ctx(ctx)

    if args.network == 'faster_rcnn_resnet50_v1b_voc':
        net.load_parameters('./model_fasterrcnn/faster_rcnn_resnet50_v1b_voc_best.params',ctx=ctx)
        print('use faster_rcnn_resnet50_v1b_voc to extract feature')
    else:
        net.load_parameters('./model_resnet101/faster_rcnn_resnet101_v1d_voc_best.params',ctx=ctx)
        print('use faster_rcnn_resnet101_v1d to extract feature')

    picpath = "test_pics"
    time_start = 0
    for i in os.listdir(picpath):
        frame = os.path.join(picpath + '/' + i)
        savepath = "result_pics/" + i.split('.')[0] + '_result.jpg'      # xxx_result.jpg

    # frame = '/home/Safety-Helmet-Wearing-Dataset-master/image/20200403131100_2764.jpg'
        parttime = time.time()
        # x, orig_img = data.transforms.presets.rcnn.load_test(frame, short=args.short)
        x, orig_img = presets.rcnn.load_test(frame, short=args.short)
        x = x.as_in_context(ctx)
        box_ids, scores, bboxes = net(x)
        ax = utils.viz.cv_plot_bbox(orig_img, bboxes[0], scores[0], box_ids[0], class_names=net.classes,thresh=args.threshold)
        time_start += time.time()-parttime
        # cv2.imshow('image', orig_img[...,::-1])
        # cv2.waitKey(0)
        cv2.imwrite(savepath, orig_img[...,::-1])
        # cv2.destroyAllWindows()
    # totaltime = time.time()-time_start
    print("time is :", time_start)`

**Comment by Aktcob** [comment]
cause the number of VOC dataset labels is 20+1(bg), and ur dataset is 4+1(bg) 

u can change this line net = model_zoo.get_model(args.network, pretrained=False) to net = model_zoo.get_model(args.network, pretrained_base=True, pretrained=False). And finetune it.

**Comment by MASON1211** [comment]
I am having the same issue
